imap_processing/mag/l1a/mag_l1a.py

```diff
@@ -81,54 +148,120 @@ def process_packets(mag_l0_list: list[MagL0]) -> dict[str, list[MagL1a]]:
             mag_l0.VECTORS, total_primary_vectors, total_secondary_vectors
         )
 
-        # Primary sensor is MAGO (most common expected case)
-        if mag_l0.PRI_SENS == 0:
-            mago_l1a = MagL1a(
+        primary_timestamped_vectors = MagL1a.calculate_vector_time(
+            primary_vectors, mag_l0.PRI_VECSEC, primary_start_time
+        )
+        secondary_timestamped_vectors = MagL1a.calculate_vector_time(
+            secondary_vectors, mag_l0.SEC_VECSEC, secondary_start_time
+        )
+
+        # Sort primary and secondary into MAGo and MAGi by 24 hour chunks
+        # TODO: Individual vectors should be sorted by day, not the whole packet
+        mago_is_primary = mag_l0.PRI_SENS == 0
+
+        mago_day = primary_day if mago_is_primary else secondary_day
```

Comment:
Yes, exactly. Eventually I need to split the individual vectors down by day as well, which should involve a minor change to this code. I was planning on leaving it, because there is some weirdness with time adjustment that I need to implement as well, so I didn't want to write a bunch of code and end up not using it.